taskRequests/script.js

```diff
@@ -84,6 +92,174 @@ function openTaskDetails(id) {
+  label.classList.add('checkbox-label');
+  label.appendChild(document.createElement('br'));
+  group.appendChild(label);
+}
+function addSortByIcon(name, id, groupName, iconPathAsc, iconPathDesc) {
+  const group = document.getElementById(groupName);
+
+  const containerAsc = createSortContainer(id, name, iconPathAsc, 'asc');
+  group.appendChild(containerAsc);
+
+  const containerDesc = createSortContainer(
+    id + '-desc',
+    name,
+    iconPathDesc,
+    'desc',
+  );
+  group.appendChild(containerDesc);
+}
+
+function sortModalButtons() {
+  const assigneeAsc = document.getElementById(ASSIGNEE_COUNT);
+  const assigneeDesc = document.getElementById(ASSIGNEE_Desc);
+  const createTimeAsc = document.getElementById(CREATED_TIME);
+  const createTimeDesc = document.getElementById(CREATED_TIME_Desc);
+
+  const sortModalButtons = [
+    assigneeAsc,
+    assigneeDesc,
+    createTimeAsc,
+    createTimeDesc,
+  ];
+  let selectedButton = null;
+  function toggleSortModal() {
+    sortModal.classList.toggle('hidden');
+  }
+
+  function selectButton(button) {
+    if (selectedButton === button) {
+      selectedButton.classList.remove('selected');
+      selectedButton = null;
+      toggleSortModal();
+    } else {
+      if (selectedButton) {
+        selectedButton.classList.remove('selected');
+      }
+      selectedButton = button;
+      selectedButton.classList.add('selected');
+      toggleSortModal();
+    }
+  }
+
+  sortModalButtons.forEach((button) => {
+    if (button) {
+      if (!button.hasEventListener) {
+        button.hasEventListener = true;
+        button.addEventListener('click', () => {
+          selectButton(button);
+        });
+      }
+    }
+  });
+}
+
+function createSortContainer(id, name, iconPath, sortOrder) {
+  const container = document.createElement('div');
+  container.classList.add('sort-container', sortOrder);
+
+  container.id = id;
+
+  const nameSpan = document.createElement('span');
+  nameSpan.textContent = name;
+  const label = document.createElement('label');
+  label.appendChild(nameSpan);
+
+  const icon = document.createElement('img');
+  icon.src = iconPath;
+  icon.alt = name + ' ' + sortOrder + ' icon';
+  icon.classList.add('sort-icon', sortOrder);
+
+  label.appendChild(icon);
+  label.classList.add('sort-label');
+
+  container.appendChild(label);
+
+  return container;
+}
+
+function populateStatus() {
+  const statusList = [
+    { name: 'Approved', id: 'APPROVED' },
+    { name: 'Pending', id: 'PENDING' },
+    { name: 'Denied', id: 'DENIED' },
+  ];
+  const requestList = [
+    { name: 'Assignment', id: 'APPROVED' },
+    { name: 'Creation', id: 'PENDING' },
+  ];
+
+  for (let i = 0; i < statusList.length; i++) {
+    const { name, id } = statusList[i];
+    addCheckbox(name, id, 'status-filter');
+  }
+  for (let i = 0; i < requestList.length; i++) {
+    const { name, id } = requestList[i];
+    addCheckbox(name, id, 'status-request');
+  }
```

Comment:
Can we use `statusList.map((id, item) => {...})` here?

Reply:
yea i used map in my recent commit